### PR TITLE
Recipe for plotting dicts

### PR DESCRIPTION
**Pull request by yha**
See #2123.

### CODE REVIEW

File: src/series.jl [hunk 1]
```diff
@@ -352,7 +352,12 @@ end
     end
 end
 
+# Dicts with Real values are plotted as bars
 
+@recipe function f(d::AbstractDict{T,<:Real}) where T
+    seriestype --> :bar
+    x, y = collect(keys(d)), collect(values(d))
```
Comment (mkborregaard) [inline src/series.jl]: I'd prefer sorting by key, unless we're dealing with an `OrderedDict`.

**Comment by mkborregaard** [comment]
I think this makes sense when T is a String, but I think a Dict(Number, Number) could be many other things than counts?
Maybe just 
```julia
@recipe function f(d::AbstractDict{T,<:Real}) where T
    xs = keys(d) |> collect |> sort
    xs, [d[x] for x in xs]
end
```
and let the user choose to display this as a bar?

**Comment by mkborregaard** [comment]
See also https://github.com/JuliaPlots/Plots.jl/issues/2123#issuecomment-519598784

**Comment by yha** [comment]
I'm not sure sorting automatically is a good idea. An `AbstractDict` may or may not have a specific ordering depending on the subtype (if it's not specifically an `OrderedDict` it might still be ordered). 
If we make a new `counts` recipe it can do `plot(sort(countmap(...)))`.

**Comment by mkborregaard** [comment]
That's also fine with me

**Comment by yha** [comment]
I suppose if we don't default to a bar plot and just extract the keys and values from the dict, we might as well do this for all types of dict, so you can plot, e.g.,
```
plot(OrderedDict("a"=>Date(2000),"b"=>Date(2001)))
```
?

**Comment by mkborregaard** [comment]
I think so

**Comment by mkborregaard** [comment]
I've often thought there are other places it would be nice to accept Dicts. Like for colours of groups.

**Pull request merged**